src/panels/developer-tools/event/developer-tools-event.ts

```diff
@@ -92,43 +91,31 @@ class HaPanelDevEvent extends LitElement {
   }
 
   private _parsedJSON() {
-    return this._computeParsedEventData(this.eventData);
+    try {
+      return this._eventData.trim() ? (load(this._eventData) as any) : {};
+    } catch (err) {
+      return ERROR_SENTINEL;
+    }
   }
 
   private _validJSON() {
-    return this._computeValidJSON(this._parsedJSON());
+    return this._parsedJSON() !== ERROR_SENTINEL;
   }
 
   private _eventSelected(ev) {
-    this.eventType = ev.detail.eventType;
+    this._eventType = ev.detail.eventType;
   }
 
   private _eventTypeChanged(ev) {
-    this.eventType = ev.target.value;
-  }
-
-  private _computeParsedEventData(eventData) {
-    try {
-      return eventData.trim() ? load(eventData) : {};
-    } catch (err) {
-      return ERROR_SENTINEL;
-    }
-  }
-
-  private _computeDocumentationUrl(hass) {
-    return documentationUrl(hass, "/docs/configuration/events/");
-  }
-
-  private _computeValidJSON(parsedJSON) {
-    return parsedJSON !== ERROR_SENTINEL;
+    this._eventType = ev.target.value;
   }
 
   private _yamlChanged(ev) {
-    this.eventData = ev.detail.value;
+    this._eventData = ev.detail.value;
```

Comment:
The `yaml-editor` will actually give you info it is valid, `ev.detail.isValid` so you can remove all the checks in this file.

Reply:
Also, `ev.detail.value` is not a string, but JSON